cmake/Corrosion.cmake

```diff
@@ -805,13 +805,29 @@ function(_add_cargo_build out_cargo_build_out_dir)
         corrosion_add_target_local_rustflags("${target_name}" "$<$<NOT:${explicit_linker_defined}>:${rustflag_linker_arg}>")
     endif()
 
+    # Unset `VSTEL_MSBuildProjectFullPath` environment variable when:
+    #   - Building with MSVC
+    #   - Using the 32-bit target platform
+    #   - The host system is not 32-bit
+    #   - The Rust version is greater or equal to 1.83
+    #   - (NOTE: If the issue is fixed in a newer version of Rust, please add upper bound to the version)
```

Comment:
I added version check because the issue was only for 1.83: https://github.com/corrosion-rs/corrosion/issues/599#issuecomment-2569121146